src/components/MarketTableVirtualized/MarketsTableVirtualized.tsx

```diff
@@ -16,65 +15,22 @@ import { SparkLineCell } from './SparkLineCell'
 import { TradeButtonCell } from './TradeButtonCell'
 import { VolumeCell } from './VolumeCell'
 
+import { AssetList } from '@/components/AssetSearch/components/AssetList'
+import { GroupedAssets } from '@/components/MarketTableVirtualized/GroupedAssets'
+import { InfiniteTable } from '@/components/ReactTable/InfiniteTable'
 import { Text } from '@/components/Text'
-import { defaultLongPressConfig, longPressSx } from '@/constants/longPress'
-import { isMobile as isMobileApp } from '@/lib/globals'
-import { vibrate } from '@/lib/vibrate'
 import { useFetchFiatAssetMarketData } from '@/state/apis/fiatRamps/hooks'
 import { breakpoints } from '@/theme/theme'
 
-const ROW_HEIGHT = 70
-
-const tableSizeSx = { base: 'sm', md: 'md' }
-const gridTemplateColumnsSx = {
-  base: '70% 30%',
-  md: '2fr 1fr 1fr 1fr 1fr 80px',
-}
-
-// Hide virtual list container scrollbar across all major browsers
-const tableContainerSx = {
-  '&::-webkit-scrollbar': {
-    display: 'none',
-  },
-  '-ms-overflow-style': 'none',
-  scrollbarWidth: 'none',
-  height: isMobileApp ? 'calc(100vh - 72px)' : '100vh',
-}
-
-const headerGridSx = {
-  display: 'grid',
-  gridTemplateColumns: gridTemplateColumnsSx,
-  width: 'full',
-  gap: '4px',
-}
-
-const gridSx = {
-  display: 'grid',
-  gridTemplateColumns: gridTemplateColumnsSx,
-  width: '100%',
-  gap: 4,
-  alignItems: 'center',
-  '& td': {
-    base: {
-      paddingRight: 4,
-      paddingLeft: 4,
-      paddingEnd: 4,
-      paddingStart: 4,
-    },
-    sm: {
-      paddingRight: '0!important',
-      paddingLeft: '0!important',
-      paddingEnd: '0!important',
-      paddingStart: '0!important',
-    },
-  },
-  ...longPressSx,
-}
-
 type MarketsTableVirtualizedProps = {
   rows: Asset[]
-  onRowClick: (arg: Row<Asset>) => void
-  onRowLongPress?: (arg: Row<Asset>) => void
+  onRowClick: (asset: Asset) => void
+  onRowLongPress?: (asset: Asset) => void
+}
+
+type AssetWithRelatedAssetIds = Asset & {
+  isGrouped?: boolean
+  relatedAssetIds?: string[]
```

Comment:
- Shouldn't `relatedAssetIds` live alongside `Asset` itself? 
- `isGrouped` makes me think this should probably be named something along the lines of asset group